lib/trino-filesystem/src/main/java/io/trino/filesystem/tracing/TracingInputFile.java

```diff
@@ -85,10 +86,17 @@ public long length()
     public Instant lastModified()
             throws IOException
     {
+        // skip tracing if lastModified is cached, but delegate anyway
```

Comment:
I feel like this requires a bit more explanation still. Does having tracing leads to too many spans? Why this method is special

Reply:
This is the same thing as `io.trino.filesystem.tracing.TracingInputFile#length`
Every TrinoInputFile caches lastModified and length, we skip recording cached accesses because tests rely on on being able to trace non-cached filesystem calls.